server/portal/apps/auth/models.py

```diff
@@ -71,11 +72,25 @@ def client(self):
         :return: Tapis client using refresh token.
         :rtype: :class:Tapis
         """
-        return Tapis(base_url=getattr(settings, 'TAPIS_TENANT_BASEURL'),
-                     client_id=getattr(settings, 'TAPIS_CLIENT_ID'),
-                     client_key=getattr(settings, 'TAPIS_CLIENT_KEY'),
-                     access_token=self.access_token,
-                     refresh_token=self.refresh_token)
+        client = Tapis(base_url=getattr(settings, 'TAPIS_TENANT_BASEURL'),
+                       client_id=getattr(settings, 'TAPIS_CLIENT_ID'),
+                       client_key=getattr(settings, 'TAPIS_CLIENT_KEY'),
+                       access_token=self.access_token,
+                       refresh_token=self.refresh_token)
+
+        with transaction.atomic():
+            if self.expired:
+                try:
+                    client.refresh_tokens()
```

Comment:
should this just be self.refresh_tokens (i.e. TapisOAuthToken.refresh_tokens) or is the another way to avoid having  lines 89-91 being duplicated with 102-104.  I like how you did it here so was first thinking of suggesting removing the TapisOAuthToken.refresh_tokens but see that is used by jwt_auth.py